wireapp/wire-ios-mono
# fix: duplicate 1-1 conversations - WPB-10260

### Issue

**Note:** this PR is based on #2017, so preferably start with #2017 first.

Due to some race condition, an user could end up with duplicate proteus team 1-1 conversation. This PR aims to fix this, by showing only one **same on all platforms** 1-1 conversation.

* Fix one to one migration that was introduced in adding the last rule to choose the 1-1 conversation (sorted qualified id)
* Add coredata migration post action to latest db migration to reset the 1-1 conversation if it was wrongly set before AND merge all messages into the conversation to prepare for MLS migration.
* Remove PatchApplicator (unused)



### Testing

* Use production account where duplicate 1-1 conversation exist
* run the migration
* See messages have been merged into one conversation

---

### Checklist
